.github/workflows/docs-preview.yml

```diff
@@ -25,23 +25,32 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0
+          # The commit to checkout. We want our actual commit, and not the result of merging the PR to the target.
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+          # Include previous commit for master history comparisons.
+          fetch-depth: 2
 
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: "18"
+          node-version: "22"
 
       - name: Enable Corepack
         run: corepack enable
 
+      - name: Install Foundry
```

Comment:
im surprised docs preview needs this

Reply:
They rebuild `l1-contracts` under certain circumstances. I've asked devrel to review this task, seems like it previously relied in cache and dreams